### PR TITLE
PAYARA-3261 OpenTracing 1.3

### PR DESCRIPTION
**Pull request by Pandrex247**
Adds OpenTracing instrumentation to Rest Client

### CODE REVIEW

File: core-client/src/main/java/org/glassfish/jersey/client/internal/inject/PrimitiveToStringInserter.java [hunk 1]
```diff
@@ -50,9 +50,8 @@
  * @author Paul Sandoz
  * @author Marek Potociar (marek.potociar at oracle.com)
  * @author Gaurav Gupta (gaurav.gupta@payara.fish)
- *
  */
-final class PrimitiveValueOfInserter implements ParameterInserter<Number, String> {
```
Comment (MattGill98) [inline core-client/src/main/java/org/glassfish/jersey/client/internal/inject/PrimitiveToStringInserter.java:55]: Is anything dependent on this name change?

File: ext/microprofile-rest-client/src/main/java/org/glassfish/jersey/microprofile/rest/client/OpenTracingInstrumentation.java [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+                            // it to the skip list after determining if it has its own path
+                            jaxrsMethodEndpoint = getMethodLevelPathValueIfPresentUsingConfig(config, method,
+                                    annotatedClassCanonicalName, annotatedMethodName, jaxrsMethodEndpoint);
+
+                            shouldSkipTracing = true;
+                        }
+                    } else {
+                        // If we didn't find a config override, check if the class is annotated and use the value
+                        // from that if it is
+                        Traced tracedAnnotation = restClientInterface.getAnnotation(Traced.class);
```
Comment (MattGill98) [inline ext/microprofile-rest-client/src/main/java/org/glassfish/jersey/microprofile/rest/client/OpenTracingInstrumentation.java:164]: This is very similar code to the annotation reader in the main codebase. Is there anything that can be combined?

File: core-client/src/main/java/org/glassfish/jersey/client/internal/inject/PrimitiveToStringInserter.java (hunk 1 above)
Comment (Pandrex247) [reply core-client/src/main/java/org/glassfish/jersey/client/internal/inject/PrimitiveToStringInserter.java:55]: Not as far as I can tell

File: ext/microprofile-rest-client/src/main/java/org/glassfish/jersey/microprofile/rest/client/OpenTracingInstrumentation.java (hunk 2 above)
Comment (Pandrex247) [reply ext/microprofile-rest-client/src/main/java/org/glassfish/jersey/microprofile/rest/client/OpenTracingInstrumentation.java:164]: Not easily - I can't depend on stuff in Jersey from Payara, and some of the stuff in Payara depends on other stuff in Payara.

**Review by MattGill98 — APPROVED** [review]

**Pull request merged**